In [1]:
import gym
import tensorflow as tf
from tensorflow import keras
import numpy as np
from collections import deque

We will solve the CartPole environment from Open-AI Gym using Deep Q-Learning. First of all, we load the environemnt.

In [2]:
env = gym.make("CartPole-v1")

We check the sizes of the action and observation spaces.

In [3]:
print(f"Observation space: {env.observation_space}")
print(f"Action space: {env.action_space}")

Observation space: Box([-4.8000002e+00 -3.4028235e+38 -4.1887903e-01 -3.4028235e+38], [4.8000002e+00 3.4028235e+38 4.1887903e-01 3.4028235e+38], (4,), float32)
Action space: Discrete(2)


The meaning for the observation space is:

| Num | Observation           | Min                  | Max                |
|-----|-----------------------|----------------------|--------------------|
| 0   | Cart Position         | -4.8                 | 4.8                |
| 1   | Cart Velocity         | -Inf                 | Inf                |
| 2   | Pole Angle            | ~ -0.418 rad (-24°)  | ~ 0.418 rad (24°)  |
| 3   | Pole Angular Velocity | -Inf                 | Inf                |